maintenance_product/models/maintenance_equipment.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2022 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import api, fields, models
+
+
+class MaintenanceEquipment(models.Model):
+    _inherit = "maintenance.equipment"
+
+    product_id = fields.Many2one(
+        comodel_name="product.product",
+        string="Product",
+        tracking=True,
+        domain="[('categ_id', '=', product_category_id)]",
+    )
+    product_category_id = fields.Many2one(
+        comodel_name="product.category", related="category_id.product_category_id"
+    )
+
+    @api.onchange("product_id")
+    def _onchange_product_id(self):
+        """If product is set, equipment name, seller, seller ref and cost defaults
+        to product ones"""
```

Comment:
Proper PEP257 docstring is:

```suggestion
        """If product is set, equipment name, seller, seller ref and cost defaults
        to product ones.
        """
```